### PR TITLE
feat: add shards and parallel

### PR DESCRIPTION
**Pull request by bhavsarpratik**

### CODE REVIEW

File: multimodal-search-tirg/flow-index.yml [hunk 1]
```diff
@@ -5,9 +5,11 @@ pods:
   read:
     uses: pods/image-read.yml
     read_only: true
+    parallel: $PARALLEL
```
Comment (JoanFM) [inline multimodal-search-tirg/flow-index.yml:8]: I would avoid the parallel here

File: multimodal-search-tirg/flow-index.yml [hunk 2]
```diff
@@ -5,9 +5,11 @@ pods:
...
   normalize:
     uses: pods/normalize.yml
     read_only: true
+    parallel: $PARALLEL
```
Comment (JoanFM) [inline multimodal-search-tirg/flow-index.yml:12]: and also here

File: multimodal-search-tirg/flow-index.yml [hunk 3]
```diff
@@ -5,9 +5,11 @@ pods:
...
   encode:
     uses: jinaai/hub.executors.encoders.image.tirg
```
Comment (JoanFM) [inline multimodal-search-tirg/flow-index.yml:14]: there is no parallelization here, why is it failing here?

**Comment by bhavsarpratik** [comment]
@JoanFM Please check and merge for your test

**Pull request merged**